SwiftCompilerSources/Sources/Optimizer/FunctionPasses/ObjectOutliner.swift

```diff
@@ -314,23 +329,25 @@ private func constructObject(of allocRef: AllocRefInstBase,
   }
   let globalBuilder = Builder(staticInitializerOf: global, context)
 
-  // Create the initializers for the tail elements.
-  let numTailTupleElems = allocRef.numStoresPerTailElement
-  if numTailTupleElems > 1 {
-    // The elements are tuples: combine numTailTupleElems elements to a single tuple instruction.
-    for elementIdx in 0..<allocRef.numTailElements! {
-      var tupleElems = [Value]()
-      for tupleIdx in 0..<numTailTupleElems {
-        let store = storesToTailElements[elementIdx * numTailTupleElems + tupleIdx]
-        tupleElems.append(cloner.clone(store.source as! SingleValueInstruction))
+  if !storesToTailElements.isEmpty {
+    // Create the initializers for the tail elements.
+    let numTailTupleElems = allocRef.numStoresPerTailElement
+    if numTailTupleElems > 1 {
+      // The elements are tuples: combine numTailTupleElems elements to a single tuple instruction.
+      for elementIdx in 0..<allocRef.numTailElements! {
+        var tupleElems = [Value]()
+        for tupleIdx in 0..<numTailTupleElems {
+          let store = storesToTailElements[elementIdx * numTailTupleElems + tupleIdx]
+          tupleElems.append(cloner.clone(store.source as! SingleValueInstruction))
+        }
```

Comment:
Drive-by comment:

Since tupleElems has a known number of elements, it would be more efficient (and arguably clearer/easier to read) to use map:

```suggestion
        let tupleElems = (0..<numTailTupleElems).map { tupleIdx in
          let store = storesToTailElements[elementIdx * numTailTupleElems + tupleIdx]
          return cloner.clone(store.source as! SingleValueInstruction)
        }
```

Not a big deal but I just happened to notice it :)